server/src/main/java/com/defold/extender/services/CocoaPodsService.java

```diff
@@ -0,0 +1,625 @@
+                String pod = line.trim().replace("- ", "").replace(":", "").replace("\"","");
+                pods.add(pod);
+            }
+        }
+
+        File podsDir = new File(workingDir, "Pods");
+        List<PodSpec> specs = new ArrayList<>();
+        Map<String, PodSpec> specsMap = new HashMap<>();
+        for (String pod : pods) {
+            // 'GoogleUtilities/Environment (7.10.0)'  -> 'GoogleUtilities/Environment'
+            String podnameparts[] = pod.replaceFirst(" \\(.*\\)", "").split("/");
+            String mainpodname = podnameparts[0];
+            if (!specsMap.containsKey(mainpodname)) {
+                // 'GoogleUtilities/Environment (7.10.0)'   ->   'GoogleUtilities --version=7.10.0'
+                String args = pod.replaceFirst("/.*?\\s", " ").replace("(", "--version=").replace(")", "");
+                String cmd = "pod spec cat " + args;
+                String specJson = execCommand(cmd).replace(cmd, "");
+
+                specsMap.put(mainpodname, createPodSpec(specJson, podsDir));
+            }
+
+            PodSpec mainpod = specsMap.get(mainpodname);
+            if (podnameparts.length == 1) {
+                specs.add(mainpod);
+            }
+            else {
+                String subspecname = podnameparts[1];
+                for (PodSpec subspec : mainpod.subspecs) {
+                    if (subspec.name.equals(subspecname)) {
+                        specs.add(subspec);
+                        break;
+                    }
+                }
+            }
+        }
+
+        return specs;
+
+    }
+
+    /**
+     * Entry point for Cocoapod dependency resolution.
+     * @param jobDirectory Root directory of the job to resolve
+     * @param platform Which platform to resolve pods for
+     * @return ResolvedPods instance with list of pods, install directory etc
+     */
+    public ResolvedPods resolveDependencies(File jobDirectory, String platform) throws IOException, ExtenderException {
+        if (!platform.contains("ios") && !platform.contains("osx")) {
+            throw new ExtenderException("Unsupported platform " + platform);
+        }
+
+        long methodStart = System.currentTimeMillis();
+        LOGGER.info("Resolving Cocoapod dependencies");
+
+        File workingDir = new File(jobDirectory, "CocoaPodsService");
+        File frameworksDir = new File(workingDir, "frameworks");
+        workingDir.mkdirs();
+
+        String platformVersion = createMainPodFile(jobDirectory, workingDir, platform);
+        List<PodSpec> pods = installPods(workingDir);
+        copyPodFrameworks(pods, frameworksDir);
+        
+        // dumpDir(jobDirectory, 0);
+
+        MetricsWriter.metricsTimer(meterRegistry, "gauge.service.cocoapods.get", System.currentTimeMillis() - methodStart);
+
+        ResolvedPods resolvedPods = new ResolvedPods();
+        resolvedPods.pods = pods;
+        resolvedPods.platformVersion = platformVersion;
```

Comment:
As Alexey mentioned, could be good with "platformMinVersion"

Reply:
👍